src/Symfony/Component/Translation/Catalogue/DiffOperation.php

```diff
@@ -30,7 +30,7 @@ protected function processDomain($domain)
         );
 
         foreach ($this->source->all($domain) as $id => $message) {
-            if ($this->target->has($id, $domain)) {
+            if (!$this->target->has($id, $domain)) {
```

Comment:
this looks wrong to me. The original code is right: if the source has it but the target does not, it is an obsolete key. your change is doing precisely the opposite